YapDatabaseExtensions/Common/YapDatabaseExtensions.swift

```diff
@@ -229,6 +252,21 @@ public protocol ValueMetadataPersistable: Persistable {
     var metadata: MetadataType { get }
 }
 
+/**
+A generic protocol which extends Persistable. It allows types to
+expose a metadata value type for use in YapDatabase, which will 
+be set automatically by YapDatabaseExtensions read functions.
+
+The metadata value must conform to Saveable.
+*/
+public protocol ExternalValueMetadataPersistable: Persistable {
+    typealias MetadataType: Saveable
+
+    /// The metadata value for this Persistable type.
+    var metadata: MetadataType { get set }
+}
+
```

Comment:
I'd make the metadata optional, so that writes without the metadata don't update the metadata.